wdh_server/app.py

```diff
@@ -60,6 +65,36 @@ def author(handle):
             self_uri=url_for("author", handle=author.handle), author=author)
 
 
+@app.route('/search')
+def search():
+    term = request.args.get("term", default="").lower()
+    category = request.args.get("category", default="")
+
+    if not term or not category:
+        print("return form")
+        return render_template("search.html",
+                               self_uri=url_for("search"),
+                               title="Search",
+                               category_author=SEARCH_CATEGORY_AUTHOR,
+                               category_article=SEARCH_CATEGORY_ARTICLE)
+
+    if category == SEARCH_CATEGORY_ARTICLE:
+        articles = [Article.from_dict(aid, article)
+                    for aid, article in enumerate(_retrieve("articles"))]
+        articles = [a for a in articles if term in a.title.lower()]
+        return render_template("articles.html", title="Search result: \"{}\" ({})".format(term, category),
+            self_uri=request.url, articles=articles)
+
+    if category == SEARCH_CATEGORY_AUTHOR:
+        authors = [Author.from_dict(handle, details) for handle, details
+                   in _retrieve("authors").items()]
+        authors = [a for a in authors if term in a.display_name.lower()]
```

Comment:
hier wird doppelt iteriert - daher habe ich's in 85a4289c7c5409ae8942c00916f14a5e7210cac9 umgeschrieben

eine Alternative zur dortigen Lösung wäre, die erste List Comprehension durch eine Generator Expression zu ersetzen